java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4039,6 +4039,19 @@ public void xreadgroup_return_failures(BaseClient client) {
                                         .get());
         assertInstanceOf(RequestException.class, executionException.getCause());
 
+        // group doesn't exists, throws a request error with "NOGROUP"
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> client.xreadgroup(Map.of(key, timestamp_1_1), "not_a_group", consumerName).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+        assertTrue(executionException.getMessage().contains("NOGROUP"));
+
+        // consumer doesn't exists, returns an empty response
```

Comment:
Just FYI, the response isn't empty because the consumer doesn't exist, its empty because we have not read anything from the stream with ">" and thus there is nothing in the Pending Entries List. Just wanted to clarify, this test is fine as is though

Reply:
Thanks - it'll create the consumer automagically.  I should update this comment.